### PR TITLE
Map the PV & PVC endpoints of pykube

### PR DESCRIPTION
**Pull request by avaczi**
Theoretically, for our purposes PVC would be enough.  Once I was there, I added PV, too.

BTW, as the mapping looks very predictable, there is maybe an easier generic solution, that does not make it necessary to add the endpoints one by one.

### CODE REVIEW

**Comment by mohabusama** [comment]
I don't think those changes are enough, since you need to update ``cluster.py`` to actually sync the new entities.

File: zmon_agent/discovery/kubernetes/kube.py [hunk 1]
```diff
@@ -62,3 +62,9 @@ def get_endpoints(self, namespace=DEFAULT_NAMESPACE) -> pykube.query.Query:
 
     def get_ingresses(self, namespace=DEFAULT_NAMESPACE) -> pykube.query.Query:
         return pykube.Ingress.objects(self.client).filter(namespace=namespace)
+
+    def get_persistentvolumes(self, namespace=DEFAULT_NAMESPACE) -> pykube.query.Query:
```
Comment (avaczi) [inline zmon_agent/discovery/kubernetes/kube.py]: In theory, namespace is not used here, but the `get_all()` call at https://github.com/avaczi/zmon-agent-core/blob/cebf24ea0ba0cb2b970b43c99a1581b1734c2d4c/zmon_agent/discovery/kubernetes/cluster.py#L134 assumes it's there

Comment (mohabusama) [reply zmon_agent/discovery/kubernetes/kube.py]: you can use ``**kwargs`` if they are not needed.

File: zmon_agent/discovery/kubernetes/kube.py [hunk 2]
```diff
@@ -62,3 +62,9 @@ def get_endpoints(self, namespace=DEFAULT_NAMESPACE) -> pykube.query.Query:
...
+        return pykube.PersistentVolume.objects(self.client).filter()
```
Comment (mohabusama) [inline zmon_agent/discovery/kubernetes/kube.py]: May be using ``.all()`` iso ``.filter``

File: zmon_agent/discovery/kubernetes/cluster.py [hunk 3]
```diff
@@ -543,6 +545,16 @@ def get_postgresql_cluster_members(kube_client, cluster_id, alias, region, infra
         pod_namespace = obj['metadata']['namespace']
         service_dns_name = '{}.{}.svc.cluster.local'.format(labels['version'], pod_namespace)
 
+        # unfortunately, there appears to be no way of filtering these on the server side :(
+        pvc_name = obj['spec']['volumes'][0]['persistentVolumeClaim']['claimName']  # assume only one PVC
```
Comment (mohabusama) [inline zmon_agent/discovery/kubernetes/cluster.py]: I suggest wrapping in ``try..except`` in order not to fail the entire thing if some pvc/pv is not as expected.

File: zmon_agent/discovery/kubernetes/kube.py (hunk 2 above)
Comment (avaczi) [reply zmon_agent/discovery/kubernetes/kube.py]: I wanted to keep consistent with the other functions (`get_noe()`, for example)

**Comment by mohabusama** [comment]
👍

**Comment by lmineiro** [comment]
👍

**Pull request merged**